gtsam/linear/JacobianFactor.cpp

```diff
@@ -35,14 +35,6 @@
 #include <boost/format.hpp>
 #include <boost/make_shared.hpp>
 #include <boost/array.hpp>
-#ifdef __GNUC__
```

Comment:
Same as above

Reply:
This was removed as I believe it was first introduced to surpress warnings from the previous <boost/bind.hpp> header. As we are not using that header, we can remove this ifdef.

Reply:
Even if we do use the older include, I believe the later versions of boost have fixed the associated warnings so there is no need to suppress them.

Reply:
You don't need to respond to each of these comments regarding the same issue :) We use this review style to mark areas so we don't miss it if it needs updating.